src/quicklist.c

```diff
@@ -325,7 +325,7 @@ REDIS_STATIC void __quicklistCompress(const quicklist *quicklist,
     if (!in_depth)
         quicklistCompressNode(node);
 
-    if (depth > 2) {
+    if (depth > 2 && quicklist->len > (unsigned int)(quicklist->compress * 2)) {
```

Comment:
I don't understand this check. This seems redundant from the check at the top of the function.

Reply:
@madolson At the top of the function do not cover `quicklist->len == (unsigned int)(quicklist->compress * 2)`.
For example, if `quicklist->compress == 3` and `quicklist->len == 6`, we have six node `1 2 3 4 5 6`，after `while`，`forward` and `reverse` will be compressed again.